### PR TITLE
polygon: dex.trades: add dodoex

### PR DESCRIPTION
**Pull request by yongjun925**
Brief comments on the purpose of your changes:

*For Dune Engine V2*
I've checked that:

* [x] I tested the query on dune.com after compiling the model with dbt compile (compiled queries are written to the target directory)
* [x] I used "refs" to reference other models in this repo and "sources" to reference raw or decoded tables 
* [x] if adding a new model, I added a test
* [x] the filename is unique and ends with .sql
* [x] each sql file is a select statement and has only one view, table or function defined  
* [x] column names are `lowercase_snake_cased`

When you are ready for a review, tag duneanalytics/data-experience. We will re-open your forked pull request as an internal pull request. Then your spells will run in dbt and the logs will be avaiable in Github Actions DBT Slim CI. This job will only run the models and tests changed by your PR compared to the production project.

### CODE REVIEW

File: polygon/erc20/tokens.sql [hunk 1]
```diff
@@ -199,6 +199,1058 @@ COPY erc20.tokens (contract_address, symbol, decimals) FROM stdin;
 \\x08e6d1f0c4877ef2993ad733fc6f1d022d0e9dbf	jPLN	18
 \\xa22f6bc96f13bcc84df36109c973d3c0505a067e	jKRW	18
 \\x6a304dFdb9f808741244b6bfEe65ca7B3b3A6076	PTaUSDC	6
+\\xd6a5ab46ead26f49b03bbb1f9eb1ad5c1767974a EMON 18
```
Comment (jeff-dude) [inline polygon/erc20/tokens.sql:202]: @yongjun925 thank you for the PR. each row you added appear to have spacing issue. assuming it's a single space between each field, that won't work. copy a prior row to get the right format (i.e. line 201). that should be applied to all new rows. hope that makes sense. once changed, i can add quickly

**Review by jeff-dude — CHANGES_REQUESTED** [review]

**Comment by yongjun925** [comment]
https://dune.com/queries/1162469?start_ts_d83555=2022-08-08+00%3A00%3A00&end_ts_d83555=2022-08-11+00%3A00%3A00

@jeff-dude I'm very sorry, the files was not submitted. Please review it again. Thanks.

**Comment by yongjun925** [comment]
image

@jeff-dude How to batch add contracts created by factory contracts? If this option is checked, it seems that only one contract address has been decoding.

**Comment by jeff-dude** [comment]
> https://dune.com/queries/1162469?start_ts_d83555=2022-08-08+00%3A00%3A00&end_ts_d83555=2022-08-11+00%3A00%3A00

@antonio-mendes help on this question?

File: polygon/erc20/tokens.sql [hunk 2]
```diff
@@ -199,6 +199,1058 @@ COPY erc20.tokens (contract_address, symbol, decimals) FROM stdin;
...
+\\x1c2e7118591ec1b6122e4fe3d645706a9747b10e	ESPRO	18
+\\x2e502bf17e29fe9d59f616501a1d189e93fc676f	808TA	18
+\\x695fc8b80f344411f34bdbcb4e621aa69ada384b	NITRO	18
+\\x32934cb16da43fd661116468c1b225fc26cf9a8c	SNE	18
+\\x6b7e88b6c6051c03ec57a5e78b880999633bd047	OIL	18
+\\x9fb758d4243f016ad049c412076cfffda22c9d06	REIN	18
+\\xe13ddcdc89b7eb61eabdd90e80c1e7f888769ad9	BEASTVERSE	9
+\\x14a92603d4f4bf605e3ef206f7614a0ab4248597	FET	8
+\\xfb8a07e99450d1dc566da18a8f0e630addefdd3e	STN	18
+\\x1fd6cf265fd3428f655378a803658942095b4c4e	YELD	V2	18
```
Comment (jeff-dude) [inline polygon/erc20/tokens.sql:832]: i fixed this row locally, but don't have access to push to your branch. please put regular space between 'YELD' and 'V2' as it's meant to be on column

**Review by jeff-dude — APPROVED** [review]
this is all built. once the change requested is pushed, i can merge this in

File: polygon/erc20/tokens.sql (hunk 2 above)
Comment (yongjun925) [reply polygon/erc20/tokens.sql]: OK, it's fixed, thanks

**Comment by jeff-dude** [comment]
thank you! i'm merging this PR now. if the open question still remains around decoding, please reach out to the dune team in discord: https://discord.com/channels/757637422384283659/850326962152538122

**Pull request merged**